src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -63,186 +67,272 @@ bool FlightTaskAutoFollowTarget::activate(const vehicle_local_position_setpoint_
+		_follow_angle_rad = math::radians(get_follow_angle_setting_deg((FollowPerspective)_param_nav_ft_fs.get()));
+	}
+}
 
-		} else if (_follow_angle_rad - new_follow_angle_rad  < -M_PI_F) {
-			_follow_angle_rad = new_follow_angle_rad - M_TWOPI_F;
+void FlightTaskAutoFollowTarget::update_target_pose_filter(const follow_target_estimator_s &follow_target_estimator) {
+	const Vector3f pos_ned_est{follow_target_estimator.pos_est};
+	const Vector3f vel_ned_est{follow_target_estimator.vel_est};
 
-		} else {
-			_follow_angle_rad = new_follow_angle_rad;
-		}
+	// Check Follow target estimator's validity & timeout conditions
+	const bool target_estimator_timeout = ((follow_target_estimator.timestamp - _last_valid_target_estimator_timestamp) > TARGET_ESTIMATOR_TIMEOUT_US);
 
-		// Lowpass the angle setting to smoothly transition to a new perspective when the user makes a change.
-		// In particular this has an effect when the setting is modified by 180 degrees, in which case the drone
-		// would pass above the target without the filter. The filtering makes it so that the drone flies around
-		// the target into the new postion.
-		_follow_angle_filter.setParameters(_deltatime, FOLLOW_ANGLE_FILTER_ALPHA);
-		_follow_angle_filter.update(_follow_angle_rad);
-
-		// Wrap around 360 degrees
-		if (_follow_angle_filter.getState() > M_TWOPI_F) {
-			_follow_angle_filter.reset(_follow_angle_filter.getState() - M_TWOPI_F);
-			_follow_angle_rad = _follow_angle_rad - M_TWOPI_F;
-
-		} else if (_follow_angle_filter.getState() < -M_TWOPI_F) {
-			_follow_angle_filter.reset(_follow_angle_filter.getState() + M_TWOPI_F);
-			_follow_angle_rad = _follow_angle_rad + M_TWOPI_F;
-		}
+	// Reset last valid estimator data received timestamp
+	_last_valid_target_estimator_timestamp = follow_target_estimator.timestamp;
 
-		// Assume the target's velocity vector is its heading and use it to construct the offset vector
-		// such that drone_pos_setpoint = target_pose + offset_vector
-		if (vel_ned_est.longerThan(MINIMUM_SPEED_FOR_HEADING_CHANGE) &&
-		    vel_ned_est.longerThan(FLT_EPSILON)) {
-			// Compute offset vector relative to target position. At the same time the offset vector defines the
-			// vieweing angle of the drone
-			_target_velocity_unit_vector = Vector2f(vel_ned_est.xy()).unit_or_zero();
-		}
+	// Reset the filter if any pose filter states are not finite or estimator timed out
+	if ((!PX4_ISFINITE(_target_pose_filter.getState()(0)) || !PX4_ISFINITE(_target_pose_filter.getState()(1))
+		|| !PX4_ISFINITE(_target_pose_filter.getState()(2)) || !PX4_ISFINITE(_target_pose_filter.getRate()(0))
+		|| !PX4_ISFINITE(_target_pose_filter.getRate()(1)) || !PX4_ISFINITE(_target_pose_filter.getRate()(2))) || target_estimator_timeout) {
+		_target_pose_filter.reset(pos_ned_est, vel_ned_est);
+	}
+	else {
+		_target_pose_filter.update(_deltatime, pos_ned_est, vel_ned_est);
+	}
+}
+
+void FlightTaskAutoFollowTarget::update_rc_adjusted_follow_height(float &follow_height) {
```

Comment:
^agree with @potaito - I also would point out I generally prefer to minimize output params (that is params getting input and then output again) if it is possible to "return" them. Helps to more clearly define the input output flow.

My two cents (opinion) - I'd go for @potaito 's second option above, or input the sticks as const ref and update the follow height state (without inputting as param) - as your method name explicitly states that it is going to change that state. So you aren't lying to anyone, and it's not a side effect.

Reply:
Since 'Sticks' is a must-have consideration for this function, passing it is a good idea!

Also I'd go for the Alessandro's 2nd option since it makes everything more verbose (since the internal variables get set in the main loop), but it is a bit more tedious than your suggestion!

Reply:
Actually, I'd go for your suggested option since it's less tedious 😏